### PR TITLE
Enforce a max hand size of 7; discard cards drawn after that

### PR DESCRIPTION
**Pull request by AlexNisnevich**

### CODE REVIEW

**Comment by Dbz** [comment]
Wow seven cards! That's very small. HS has a max hand size of 10. If we're going to have a maximum hand size of 7, I wonder if the cards should be more powerful? Just thoughts to consider.

File: src/common/util/game.js [hunk 1]
```diff
@@ -321,7 +321,22 @@ function endTurn(state) {
...
+  const numCardsDiscarded = count - numCardsDrawn;
+
+  if (numCardsDrawn > 0) {
+    player.hand = player.hand.concat(player.deck.splice(0, numCardsDrawn));
+  }
+
+  times(numCardsDiscarded, () => {
+    const card = player.deck[0];
+    player.deck.splice(0, 1);
+    state = logAction(state, player, `had to discard |${card.name}| due to having a full hand of ${MAX_HAND_SIZE} cards`, {
```
Comment (Dbz) [inline src/common/util/game.js]: What do you think of having cards highlighted in the feed, and when you hover over a card, it displays it.

Comment (AlexNisnevich) [inline src/common/util/game.js]: That's already how it works! :-)

**Comment by AlexNisnevich** [comment]
I'm not sure what the max hand size should be. I just set it to 7 in the interest of not having cards overlap too much in the worst case.

It's 8 in Magic and 9 in Faeria, so perhaps it should be adjusted up a bit.

But it's a really easy tweak to change it at any point – just one line in `constants.js`.

**Comment by AlexNisnevich** [comment]
I'll keep it at 7 for this PR, but we can always raise it to 8, 9, or 10 depending on playtesting.

**Pull request merged**